src/archivist/recorder/repositories/mongo/index.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * This module is the boundary beyond which the usage of MongoDB is abstracted.
+ * Object IDs are used as opaque unique IDs.
+ */
+
+import { MongoClient, ObjectId, Binary } from 'mongodb';
+
+import RepositoryInterface from '../interface.js';
+
+import DataMapper from './dataMapper.js';
+
+export default class MongoRepository extends RepositoryInterface {
+  constructor({ database: databaseName, collection: collectionName, connectionURI }) {
+    super();
+    const client = new MongoClient(connectionURI);
```

Comment:
Shouldn't the client be a private variable?
```suggestion
    this.#client = new MongoClient(connectionURI);
```